File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int mx=INT_MIN;
        int t=0;
        for(auto i:nums){
            t+=i;
            mx=max(mx,t);
            if(t<0)
                t=0;
        }
        return mx;
    }
};

File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int mx=Integer.MIN_VALUE,cur=0;
        for(int i=0;i<nums.length;i++){
            cur+=nums[i];
            if(cur>mx)
                mx=cur;
            if(cur<0)
                cur=0;
        }
        return mx;
    }
}